src/libraries/System.Collections.Immutable/tests/Frozen/FrozenDictionaryTests.cs

```diff
@@ -392,19 +406,68 @@ public void CreateHugeDictionary_Success(int largeCount)
         }
     }
 
-    public class FrozenDictionary_Generic_Tests_int_int : FrozenDictionary_Generic_Tests<int, int>
+    public abstract class FrozenDictionary_Generic_Tests_base_for_numbers<T> : FrozenDictionary_Generic_Tests<T, T>
     {
         protected override bool DefaultValueAllowed => true;
 
-        protected override KeyValuePair<int, int> CreateT(int seed)
+        protected override KeyValuePair<T, T> CreateT(int seed)
         {
             Random rand = new Random(seed);
-            return new KeyValuePair<int, int>(rand.Next(), rand.Next());
+            return new KeyValuePair<T, T>(Next(rand), Next(rand));
         }
 
-        protected override int CreateTKey(int seed) => new Random(seed).Next();
+        protected override T CreateTKey(int seed) => Next(new Random(seed));
+
+        protected override T CreateTValue(int seed) => CreateTKey(seed);
+
+        protected abstract T Next(Random random);
+    }
+
+
+    public class FrozenDictionary_Generic_Tests_int_int : FrozenDictionary_Generic_Tests_base_for_numbers<int>
+    {
+        protected override int Next(Random random) => random.Next(); 
```

Comment:
the existing test was using `Next` overload that does not generate negative values. I decided to keep it that way